tools/traffic/table/blob_table.go

```diff
@@ -0,0 +1,125 @@
+package table
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+)
+
+// BlobTable tracks blobs written by the traffic generator. This is a thread safe data structure.
+type BlobTable struct {
+
+	// blobs contains all blobs currently tracked by the requiredReads.
+	blobs []*BlobMetadata
+
+	// size describes the total number of blobs currently tracked by the requiredReads.
+	// size may be smaller than the capacity of the blobs slice.
+	size uint
+
+	// lock is used to synchronize access to the requiredReads.
+	lock sync.Mutex
+}
+
+// NewBlobTable creates a new BlobTable instance.
+func NewBlobTable() BlobTable {
+	return BlobTable{
+		blobs: make([]*BlobMetadata, 1024),
+		size:  0,
+	}
+}
+
+// Size returns the total number of blobs currently tracked by the requiredReads.
+func (table *BlobTable) Size() uint {
+	table.lock.Lock()
+	defer table.lock.Unlock()
+
+	return table.size
+}
+
+// Get returns the blob at the specified index. Returns nil if the index is out of bounds.
+func (table *BlobTable) Get(index uint) *BlobMetadata {
+	table.lock.Lock()
+	defer table.lock.Unlock()
+
+	if index >= table.size {
+		return nil
+	}
+
+	return table.blobs[index]
+}
+
+// Add a blob to the requiredReads.
+func (table *BlobTable) Add(blob *BlobMetadata) {
+	table.lock.Lock()
+	defer table.lock.Unlock()
+
+	blob.index = table.size
+	table.blobs[table.size] = blob
```

Comment:
what if the `table.size` is out of bound?

Reply:
Fixed.

Good catch. It just so happened that the original capacity was 1024, which exceeded the number of blobs I was testing with.

I also made the initial capacity 0, thus making existing unit tests sensitive to this problem.